src/main/scala/nodes/learning/SparseLinearMapper.scala

```diff
@@ -0,0 +1,67 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package nodes.learning
+
+import breeze.linalg._
+import org.apache.spark.rdd.RDD
+import workflow.Transformer
+
+/**
+ * Computes A * x + b i.e. a linear map of data using a trained model.
+ *
+ * @param x trained model
+ * @param bOpt optional intercept to add
+ */
+case class SparseLinearMapper(
```

Comment:
That definitely sounds like a nicer situation from the user's perspective, but at the moment they would still have to manually specify a sparse vs dense gradient method. We could sort something out there too, but I'm not sure how much benefit it would add unless we start allowing more nodes to work on either sparse or dense data (kinda like how mllib does it). 

I think it's something we should open an issue for and visit at some point, I'm just concerned that it could end up similar to our attempt to abstract Float vs. Double (where for some nodes it's not that hard to do, and for others it would be a big pain).